lib/Lower/IO.cpp

```diff
@@ -295,15 +258,12 @@ lowerStringLit(Fortran::lower::AbstractConverter &converter, mlir::Location loc,
                mlir::Type ty2 = {}) {
   auto &builder = converter.getFirOpBuilder();
   auto *expr = Fortran::semantics::GetExpr(syntax);
-  auto buffer = converter.genExprValue(expr, loc);
-  auto buff = builder.create<fir::ConvertOp>(loc, ty0, buffer);
-  mlir::Type ty = buffer.getType();
-  auto lenVal = ty.cast<fir::SequenceType>().getShape()[0];
-  auto len = builder.create<mlir::ConstantOp>(
-      loc, builder.getIntegerAttr(ty1, lenVal));
+  auto str = converter.genExprValue(expr, loc);
+  auto dataLen = builder.materializeCharacter(str);
```

Comment:
Heads-up here: I want to go in a direction of creating merged at link time (ODR) globals for constant string literals. That's not done yet, but in the works...